src/compiler/checker.ts

```diff
@@ -23419,6 +23423,14 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             case SyntaxKind.JSDocFunctionType:
                 error(declaration, Diagnostics.Function_type_which_lacks_return_type_annotation_implicitly_has_an_0_return_type, typeAsString);
                 return;
+            case SyntaxKind.JSDocSignature:
+                if (noImplicitAny && isJSDocOverloadTag(declaration.parent)) {
+                    const jsDocOverloadTag = declaration.parent;
+                    const sourceFile = getSourceFileOfNode(jsDocOverloadTag);
+                    diagnostics.add(
+                        createFileDiagnostic(sourceFile, jsDocOverloadTag.pos, jsDocOverloadTag.tagName.end - jsDocOverloadTag.pos, Diagnostics.This_overload_implicitly_returns_the_type_0_because_it_lacks_a_return_type_annotation, typeAsString));

```

Comment:
I think you can pass jsDocOverLoadTag.tagName to `error` to avoid referencing pos/end and sourceFile.

Reply:
Should `@` be included in the error range?
 
```shell
   /**
     * @overload
        ~~~~~~~~
!!! error TS7010: 'overload', which lacks return-type annotation, implicitly has an 'any' return type.
     * @param {string} x
     */
```

Reply:
I don't think it matters. I'd go with tagName because the code is so much shorter.

Reply:
Oke, reverted